zebra-chain/src/transaction/serialize.rs

```diff
@@ -152,13 +150,27 @@ impl ZcashSerialize for Transaction {
                 // instead we have to interleave serialization of the
                 // ShieldedData and the JoinSplitData.
 
-                match shielded_data {
+                match sapling_shielded_data {
                     None => {
-                        // Signal no shielded spends and no shielded outputs.
+                        // Signal no sapling value balance, anchor variant,
+                        // shielded spends, shielded outputs.
+                        writer.write_compactsize(0)?;
+                        writer.write_compactsize(0)?;
+                        writer.write_compactsize(0)?;
+                        writer.write_compactsize(0)?;
+
+                        // Todo: check was going on here
+                        // zip243_deserialize_and_round_trip test needs all this 0s at the end
+                        // to pass the last assert_equal.
+                        writer.write_compactsize(0)?;
+                        writer.write_compactsize(0)?;
+                        writer.write_compactsize(0)?;
+                        writer.write_compactsize(0)?;
```

Comment:
Let's look at what the spec says here. Can you take a screenshot of the spec and post it in reply to this comment?

The spec takes priority over test code. Was the test code was also changed in this PR?